pkg/dispatcher/dispatcher.go

```diff
@@ -0,0 +1,204 @@
+package dispatcher
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/buraksezer/consistent"
+	"github.com/cespare/xxhash"
+	mapset "github.com/deckarep/golang-set/v2"
+	"github.com/openshift-online/maestro/pkg/api"
+	"github.com/openshift-online/maestro/pkg/dao"
+	logger "github.com/openshift-online/maestro/pkg/logger"
+)
+
+// Dispatcher interface outlines methods for coordinating resource status updates
+// in the context of multiple active Maestro instances. Each instance subscribes
+// to a shared topic for resource status updates.
+//
+// The dispatcher's role is to ensure that only one instance processes specific
+// resource status updates, thereby optimizing efficiency and preventing redundancy.
+type Dispatcher interface {
+	// Start initiates the dispatcher with the provided context
+	Start(ctx context.Context) error
+	// Dispatch sends resource status updates to the appropriate maestro instance based on the provided key
+	Dispatch(key string) bool
```

Comment:
what is the key indicate here, is it consumer id?

Reply:
yes, key is consumer ID.